ocaml/typing/typecore.ml

```diff
@@ -504,9 +501,17 @@ let register_allocation_value_mode mode =
   register_allocation_mode alloc_mode;
   alloc_mode
 
+let register_allocation_value_mode' mode =
```

Comment:
Terrible name. What does `'` mean? It's used in two places, and saves a single easy-to-understand line. I think you should just inline it into those places.